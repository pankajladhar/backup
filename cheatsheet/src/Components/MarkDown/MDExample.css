@import 'githubMarkdown.css';

.MarkDown:first-child .MarkDown__Divider {
    display: none;
}

.MarkDown .MarkDown__Title {
    margin: 0;
    font-weight: normal;
    line-height: 1;
}

.MarkDown .MarkDown__Info {
    margin: 0;
    padding: 10px 0 30px;
    color: #666;
}


.MDExample {
    display: flex;
    flex-direction: column;
    margin-bottom: 40px;
}

.MDExample__Body {
    display: flex;
    border: 1px #e8e8e8 solid;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    overflow: hidden;
}


.MDExample__Info h3 {
    margin: 0;
}

.MDExample__Info p {
    margin: 0;
    color: #666;
    padding-bottom: 20px;
}

.MDExample__Code, 
.MDExample__Preview {
    flex: 1;
}

.MDExample__Code {
    border-right: 1px #e8e8e8 solid;
}

.MDExample__Title {
    background: #e8e8e8;
    font-weight: bold;
    padding: 5px 20px;
}

.MDExample__Content {
    padding: 10px 15px;
}

.MDExample__Content p {
    margin: 0px;
}

.MarkDown__Divider {
    text-align: center;
    margin-bottom: 40px;
    line-height: 1;
}

.MarkDown__Divider hr {
    border-top: 1px dashed #a9a9a9;
    width: 400px;
    display: inline-block;
    margin: 0;
    background: #f3f0f0;
    line-height: 1;
}

@media (max-width: 767px) { 
    .MarkDown__Divider hr { 
        width: 280px;
    }
}