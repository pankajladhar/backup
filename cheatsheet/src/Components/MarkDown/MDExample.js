import React from 'react';
import PropTypes from 'prop-types';
import './MDExample.css'

const MDExample = (props) => {
    return (
        <div className="MDExample">
            {
                props.title && props.description &&
                <div className="MDExample__Info">
                    <h3>{props.title}</h3>
                    <p>{props.description}</p>
                </div>
            }
            <div className="MDExample__Body">
                {props.children}
            </div>
        </div>
    )
}

MDExample.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    children: PropTypes.node
}

export default MDExample;