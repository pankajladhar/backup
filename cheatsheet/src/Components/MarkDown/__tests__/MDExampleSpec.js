import React from 'react';
import renderer from 'react-test-renderer';
import MDExample from './../MDExample';

describe("<MDExample />", () => {
    it('should render properly if title and description passed', () => {
        let tree = renderer.create(
            <MDExample title="title"
                description="description" />
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })

    it('should render properly if only children passed', () => {
        let tree = renderer.create(
            <MDExample>
                <span>children</span>
            </MDExample>
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })
})