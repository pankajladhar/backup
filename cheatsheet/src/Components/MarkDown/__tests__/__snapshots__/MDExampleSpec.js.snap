// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MDExample /> should render properly if only children passed 1`] = `
<div
  className="MDExample"
>
  <div
    className="MDExample__Body"
  >
    <span>
      children
    </span>
  </div>
</div>
`;

exports[`<MDExample /> should render properly if title and description passed 1`] = `
<div
  className="MDExample"
>
  <div
    className="MDExample__Info"
  >
    <h3>
      title
    </h3>
    <p>
      description
    </p>
  </div>
  <div
    className="MDExample__Body"
  />
</div>
`;
